app/controllers/api/v1/config_templates_controller.rb

```diff
@@ -78,9 +80,23 @@ def destroy
       api :GET, "/config_templates/build_pxe_default", "Change the default PXE menu on all configured TFTP servers"
 
       def build_pxe_default
-        status, msg = ConfigTemplate.authorized(:deploy_templates).build_pxe_default(self)
+        status, msg = ProvisioningTemplate.authorized(:deploy_provisioning_templates).build_pxe_default(self)
         render :json => msg, :status => status
       end
+
+      private
+
+      def type_name_singular
+        @type_name_singular ||= resource_name
+      end
+
+      def resource_class
+        ProvisioningTemplate
+      end
+
+      def deprecated
+        ::ActiveSupport::Deprecation.warn('Config templates were renamed to provisioning templates')
```

Comment:
v1 (and v2?) should support this forever, maybe the message should say that it will be removed in v3? I wanted to hear also about other deprecations (not just API) from others, e.g. we should say that something will be dropped in Foreman 1.10 and then during release process remove deprecated methods.